src/components/Dashboard.jsx

```diff
@@ -1,7 +1,206 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { useRef } from "react";
 
 const Dashboard = () => {
-  return <div className="container">Dashboard</div>;
+  const [postList, setPostList] = useState([]);
+  const textareaRef = useRef();
+  const [postTitle, setPostTitle] = useState("");
+  const { name } = useContext(UserContext);
+
+  useEffect(() => {
+    let retrievedPosts;
+    if (localStorage.getItem("posts")) {
+      retrievedPosts = JSON.parse(localStorage.getItem("posts"));
+      setPostList(retrievedPosts);
+    }
+  }, []);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const newPost = {
+      id: new Date().getTime(),
+      title: postTitle,
+      date: new Date(),
+      author: name,
+      isPostOpen: false,
+      likeCount: 0,
+      isLiked: false,
+    };
+
+    if (postTitle) {
+      postList.push(newPost);
+      localStorage.setItem("posts", JSON.stringify(postList));
+    }
+
+    setPostTitle("");
+    textareaRef.current.focus();
+  };
+
+  const handleDelete = (id) => {
+    const deletedItem = postList.filter((item) => item.id !== id);
+    setPostList(deletedItem);
+    localStorage.setItem("posts", JSON.stringify(deletedItem));
+  };
+
+  const openCommentBox = (id) => {
+    const newPosts = [...postList];
+    const currentPost = postList.find((post) => post.id === id);
+    currentPost.isPostOpen = !currentPost.isPostOpen;
+    setPostList(newPosts);
+  };
+
+  const handleLikeCount = (id) => {
```

Comment:
we can also refactor this function code  according to this https://github.com/mkh-wtag/blog-app/pull/1/files#r1421917084 .